interview-questions.md

```diff
@@ -8,42 +8,43 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is object-oriented programming? How is it different than functional programming?
 
-Your answer:
+Your answer: objective - oriented programming is based on the concepts of objects, each with it's own properties or attributes. Functional programming mainly supports abstraction over data and abstraction over behavior. 
 
-Researched answer:
+Researched answer: n object-oriented programming data structures, or objects are defined, each with its own properties or attributes.Functional programming provides high performance in processing large data for applications. Object-oriented programming is not good for big data processing.
 
 1. What is the difference between a Float and an Integer in Ruby?
 
-Your answer:
+Your answer: An integers are whole numbers, such as 10. However, floats are numbers that contain a decimal point, such as 10.5, 10.0, or 0.25.
 
-Researched answer:
+Researched answer: 
+Integers and floats are two different kinds of numerical data. An integer (more commonly called an int) is a number without a decimal point. A float is a floating-point number, which means it is a number that has a decimal place. Floats are used when more precision is needed.
 
 1. Ruby has an implicit return. What does that mean?
 
-Your answer:
+Your answer: Ruby's implicit return means that if a return is the last expression in a path of execution, there's no need for the return keyword
 
-Researched answer:
+Researched answer: The implicit return of a Ruby method is the final statement inside the method. In other words, the method will return the result of the final line before the end keyword.
 
 1. What is a block in Ruby?
 
-Your answer:
+Your answer: Ruby rocks are little anonymous functions that can be passed into methods
 
-Researched answer:
+Researched answer: Ruby has a concept of Block. A block consists of chunks of code. You assign a name to a block. The code in the block is always enclosed within braces ({}). A block is always invoked from a function with the same name as that of the block.
 
 1. How do you extract a value in a Ruby hash?
 
-Your answer:
+Your answer: In order to exract a value in a Ruby hash, it would be after the hash name. next type the key in the square brackets in order to access the value. 
 
-Researched answer:
+Researched answer: Getting the keys of a hash at once in Ruby is possible using the keys property. It returns a new array that contains all the keys of a hash.
```

Comment:
Nice work in this section!  Some of the initial answers are a little thin, so consider expanding whenever possible as if you were talking to an interviewer at your dream job.